docs/manual/creating-a-checker.tex

```diff
@@ -1141,83 +1136,35 @@
-literal (other than \<null>) has a \refqualclass{checker/nullness/qual}{NonNull} type.
-Section~\ref{effective-qualifier} explains the meaning of implicit
-qualifiers, such as that they cannot be overridden.
-
-The implicit annotations may be specified declaratively
-(Section~\ref{creating-declarative-type-introduction}) and/or procedurally
-(Section~\ref{creating-procedurally-specifying-implicit-annotations}).
-It is easiest to specify them declaratively, when the declarative method is
-sufficiently expressive.
-
-Default annotations are distinct from implicit annotations.
-Section~\ref{creating-typesystem-defaults} describes how to specify
-default annotations.
-
-
-\subsectionAndLabel{Declaratively specifying implicit annotations}{creating-declarative-type-introduction}
-
-The \refqualclass{framework/qual}{ImplicitFor} meta-annotation indicates
-implicit annotations.
-When written on a qualifier, \refclass{framework/qual}{ImplicitFor}
-specifies the trees (AST nodes) and types for which the framework should
-automatically add that qualifier.
-
-In short, the types and trees can be
-specified via any combination of four fields in \refclass{framework/qual}{ImplicitFor}.
+\sectionAndLabel{Type factory: Type introduction rules}{creating-type-introduction}
 
-For example, consider these (partial) definitions of the
-\refqualclass{checker/nullness/qual}{Nullable} and
-\refqualclass{checker/nullness/qual}{NonNull}
-type qualifiers:
-
-\begin{Verbatim}
-  @SubtypeOf({})
-  @ImplicitFor(literals = { LiteralKind.NULL },
-               typeNames = { java.lang.Void.class })
-  @Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
-  public @interface Nullable {}
-
-  @SubtypeOf( { Nullable.class } )
-  @ImplicitFor(literals = { LiteralKind.STRING },
-               types = { TypeKind.PACKAGE, TypeKind.INT, TypeKind.BOOLEAN,
-                         TypeKind.CHAR, TypeKind.DOUBLE, TypeKind.FLOAT,
-                         TypeKind.LONG, TypeKind.SHORT, TypeKind.BYTE })
-  @Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
-  public @interface NonNull {}
-\end{Verbatim}
-
-For more details, see the Javadoc for the \refclass{framework/qual}{ImplicitFor}
-annotation, and the Javadoc for the javac classes that are linked from
-it.  You only need to understand a small amount about the javac AST, such
-as the
-\href{\TreeAPIBase{}/tree/Tree.Kind.html?is-external=true}{\code{Tree.Kind}}
-and
-\refModelclass{type}{TypeKind}
-enums.  All the information you need is in the Javadoc, and
-Section~\ref{creating-javac-tips} can help you get started.
+The annotated type of expressions and types are defined via type introduction rules in the
+type factory.  For most expressions and types, these rules are the same regardless of the type system.
+For example, the type of a method invocation expression is the return type of the invoked method
+view-point-adapted for the call site.  The framework implements these rules so that all type systems
```

Comment:
Is it "viewpoint" or "view-point"? You've used both in this PR, so we should be consistent throughout.